### PR TITLE
Fix/separation

### PR DESCRIPTION
**Pull request by QBYMKPI**
# Description

This module allows the creation and management of folders, including support for IAM bindings, organization policies, and hierarchical firewall rules. It is cloned from the CFF on the Google Cloud Platform Repository.

# How Has This Been Tested?
- [x] Apply & Destroy Examples

# Checklist:

- [x] I have run tests and documented them above
- [x] I have performed a self-review of my own code
- [x] I have updated the documentation

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -323,3 +323,548 @@ module "folder" {
 | [sink_writer_identities](outputs.tf#L45) | Writer identities created for each sink. |  |
 
 <!-- END TFDOC -->
```
Comment (ghost) [inline README.md]: Please remove the now duplicated docs left by google.

File: README.md [hunk 2]
```diff
@@ -323,3 +323,548 @@ module "folder" {
...
+
+<!-- BEGIN_TF_DOCS -->
+## Usage
+
+This Module creates a GCP Folder
```
Comment (ghost) [inline README.md:12]:
```suggestion

## Basic

This Module creates a GCP Folder.
```

File: README.md [hunk 3]
```diff
@@ -323,3 +323,548 @@ module "folder" {
...
+  description = "The Project ID where to create the Folder"
+}
+
+variable "organization_domain" {
+  type = string
+}
+```
+
+This Module creates a GCP Folder with a firewall policy
+It is possible to include yaml config files.
```
Comment (ghost) [inline README.md]: Copied from googles docs. 

```suggestion
### Firewall policy factory

In the same way as for the [organization-configuration](https://github.com/qbeyond/terraform-google-organization-configuration) module, the in-built factory allows you to define a single policy, using one file for rules, and an optional file for CIDR range substitution variables. Remember that non-absolute paths are relative to the root module (the folder where you run `terraform`).

```

File: README.md [hunk 4]
```diff
@@ -323,3 +323,548 @@ module "folder" {
...
+  priority: 1002
+  ranges:
+    - 35.235.240.0/20
+  ports:
+    tcp: ["22"]
+  target_resources: null
+  enable_logging: false
+```
+
+This Module creates a GCP Folder with a hierarchical firewall policy
```
Comment (ghost) [inline README.md:113]:
```suggestion
### Hierarchical firewall policies

This example creates a GCP Folder with a hierarchical firewall policy 
```

**Review by ghost — CHANGES_REQUESTED** [review]
Looks quite good. Please have a look at the docs from google and copy over headlines and explanation to the new auto generated docs.

File: README.md (hunk 3 above)
Comment (QBYMKPI) [reply README.md]: done

File: .terraform-docs.yml [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+  Remember that non-absolute paths are relative to the root module (the folder where you run `terraform`).
+  ```hcl
+  {{ include "examples/firewall_policies/main.tf" }}
+  {{ include "examples/firewall_policies/variables.tf" }}
+  ```
+
+  ```yaml
+  {{ include "examples/firewall_policies/configs/cidrs.yaml" }}
+  {{ include "examples/firewall_policies/configs/rules.yaml" }}
+  ```
```
Comment (ghost) [inline .terraform-docs.yml]:
```suggestion
   `configs/cidrs.yaml`
  ```yaml
  {{ include "examples/firewall_policies/configs/cidrs.yaml" }}
  ``` 
  
     `configs/rules.yaml`
  ```yaml
   {{ include "examples/firewall_policies/configs/rules.yaml" }}
  ``` 
```

**Review by ghost — CHANGES_REQUESTED** [review]

File: examples/basic/variables.tf [hunk 6]
```diff
@@ -0,0 +1,8 @@
+variable "project_id" {
+  type = string
+  description = "The Project ID where to create the Folder"
```
Comment (ghost) [inline examples/basic/variables.tf]: Folders are not created inside projects.

File: examples/firewall_policies/variables.tf [hunk 7]
```diff
@@ -0,0 +1,8 @@
+variable "project_id" {
+  type = string
+  description = "The Project ID where to create the Folder"
```
Comment (ghost) [inline examples/firewall_policies/variables.tf]: Folders are not created inside projects.

File: examples/hierarchical_firewall_policies/variables.tf [hunk 8]
```diff
@@ -0,0 +1,8 @@
+variable "project_id" {
+  type = string
+  description = "The Project ID where to create the Folder"
```
Comment (ghost) [inline examples/hierarchical_firewall_policies/variables.tf]: Folders are not created inside projects.

File: examples/iam/variables.tf [hunk 9]
```diff
@@ -0,0 +1,13 @@
+variable "project_id" {
+  type = string
+  description = "The Project ID where to create the Folder"
```
Comment (ghost) [inline examples/iam/variables.tf]: Folders are not created inside projects.

File: examples/org_policies/variables.tf [hunk 10]
```diff
@@ -0,0 +1,8 @@
+variable "project_id" {
+  type = string
+  description = "The Project ID where to create the Folder"
```
Comment (ghost) [inline examples/org_policies/variables.tf]: Folders are not created inside projects.

File: examples/sink/variables.tf [hunk 11]
```diff
@@ -0,0 +1,8 @@
+variable "project_id" {
+  type = string
+  description = "The Project ID where to create the Folder"
```
Comment (ghost) [inline examples/sink/variables.tf]: Folders are not created inside projects.

File: examples/tags/variables.tf [hunk 12]
```diff
@@ -0,0 +1,8 @@
+variable "project_id" {
+  type = string
+  description = "The Project ID where to create the Folder"
```
Comment (ghost) [inline examples/tags/variables.tf]: Folders are not created inside projects.

**Review by ghost — CHANGES_REQUESTED** [review]
Just change the description, why you need a project and it's good to go. Did you test, if you need to overwrite it in any cases and not just specific cases?

File: examples/basic/variables.tf (hunk 6 above)
Comment (QBYMKPI) [reply examples/basic/variables.tf]: removed

File: examples/firewall_policies/variables.tf (hunk 7 above)
Comment (QBYMKPI) [reply examples/firewall_policies/variables.tf]: removed

File: examples/hierarchical_firewall_policies/variables.tf (hunk 8 above)
Comment (QBYMKPI) [reply examples/hierarchical_firewall_policies/variables.tf]: removed

File: examples/iam/variables.tf (hunk 9 above)
Comment (QBYMKPI) [reply examples/iam/variables.tf]: Is needed to activate service when not impersonating Service Account. Changed variable name to billing_project_id.

File: examples/org_policies/variables.tf (hunk 10 above)
Comment (QBYMKPI) [reply examples/org_policies/variables.tf]: Is needed to activate service when not impersonating Service Account. Changed variable name to billing_project_id.

File: examples/sink/variables.tf (hunk 11 above)
Comment (QBYMKPI) [reply examples/sink/variables.tf]: Is needed to create a Bucket. Changed variable name to bucket_project_id.

File: examples/tags/variables.tf (hunk 12 above)
Comment (QBYMKPI) [reply examples/tags/variables.tf]: removed

File: README.md [hunk 13]
```diff
@@ -230,96 +159,373 @@ module "folder1" {
...
+| [google_folder.folder](https://registry.terraform.io/providers/hashicorp/google/latest/docs/data-sources/folder) | data source |
+### organization-policies.tf
+| Name | Type |
+|------|------|
+| [google_org_policy_policy.default](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/org_policy_policy) | resource |
+### tags.tf
+| Name | Type |
+|------|------|
+| [google_tags_tag_binding.binding](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/tags_tag_binding) | resource |
+<!-- END_TF_DOCS -->
```
Comment (schnaddy) [inline README.md:531]:
```suggestion
<!-- END_TF_DOCS -->

## Contribute


This module is derived from [google cloud foundation fabric module `folder` v19](https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/tree/v19.0.0/modules/folder). It is designed to be able to integrate new changes from the base repository. Refer to [guide in `terraform-google-landing-zone` repository](https://github.com/qbeyond/terraform-google-landing-zone/tree/main#updating-a-repository) for information on integrating changes.
```

File: examples/iam/variables.tf [hunk 14]
```diff
@@ -1,6 +1,6 @@
-variable "project_id" {
+variable "billing_project_id" {
   type = string
-  description = "The Project ID where to create the Folder"
+  description = "The Project ID of Billing Account"
```
Comment (ghost) [inline examples/iam/variables.tf]:
```suggestion
  description = "The Project ID to use for billing."
```

File: examples/org_policies/variables.tf [hunk 15]
```diff
@@ -1,6 +1,6 @@
-variable "project_id" {
+variable "billing_project_id" {
   type = string
-  description = "The Project ID where to create the Folder"
+  description = "The Project ID of Billing Account"
```
Comment (ghost) [inline examples/org_policies/variables.tf]:
```suggestion
  description = "The Project ID to use for billing."
```

**Review by ghost — DISMISSED** [review]

**Review by ghost — APPROVED** [review]

File: .terraform-docs.yml [hunk 16]
```diff
@@ -0,0 +1,161 @@
...
+  
+  # Google Cloud Folder Module
+
+  This module allows the creation and management of folders, including support for IAM bindings, organization policies, and hierarchical firewall rules.
+
+  ## Examples
+
+  ### Basic
+
+  This Module creates a GCP Folder
```
Comment (schnaddy) [inline .terraform-docs.yml:51]: Bei der Modulbeschreibung fehlt noch die referenz zu Cloud Foundation fabric.
"Original Module from [Cloud-Foundation-Fabric](https://github.com/GoogleCloudPlatform/cloud-foundation-fabric)"

Wollen wir den Part in die Autodoku packen oder wie beim gcs Modul die Kurzbeschreibung des Moduls und diesen Part vor die Autodoku schieben?

**Review by schnaddy — CHANGES_REQUESTED** [review]
Missing parts in ReadMe. Mention of original module and Contribute block at the end

**Pull request merged**